test/Riok.Mapperly.Tests/Mapping/ObjectPropertyTest.cs

```diff
@@ -443,4 +456,33 @@ public void ShouldIgnoreStaticConstructorAndDiagnostic()
             .HaveDiagnostic(DiagnosticDescriptors.CouldNotCreateMapping)
             .HaveAssertedAllDiagnostics();
     }
+
+    [Fact]
+    public void ShouldFlattedNestedToTargetRootMapping()
+    {
+        var mapperBody = """
+            [MapperNestedProperties(nameof(A.NestedValue)]
```

Comment:
Could you also add a test for deep nested properties. Eg. `[MapNestedProperties(new[] { nameof(A.Nested), nameof(A.Nested.Value) })]`? I'm not sure, but I think these won't work yet.
And probably also add a test for q required init property, a constructor parameter and a queryable projection mapping.